sections/video.liquid

```diff
@@ -100,22 +92,39 @@
         </span>
       </button>
       <template>
-        {%- if section.settings.video_url.type == 'youtube' -%}
-          <iframe
-            src="https://www.youtube.com/embed/{{ section.settings.video_url.id }}?enablejsapi=1"
-            class="js-youtube"
-            allow="autoplay; encrypted-media"
-            allowfullscreen
-            title="{{ section.settings.description | escape }}"
-          ></iframe>
+        {%- if section.settings.video == null -%}
+          {%- liquid
+            assign loop = ''
+            if section.settings.enable_video_looping
+              assign loop = '&loop=1&playlist=' | append: video_id
+            endif
+          -%}
+          {%- if section.settings.video_url.type == 'youtube' -%}
+            <iframe
+              src="https://www.youtube.com/embed/{{ video_id }}?enablejsapi=1&autoplay=1{{ loop }}"
+              class="js-youtube"
+              allow="autoplay; encrypted-media"
+              allowfullscreen
+              title="{{ section.settings.description | escape }}"
+            ></iframe>
+          {%- else -%}
+            <iframe
+              src="https://player.vimeo.com/video/{{ video_id }}?autoplay=1{{ loop }}"
```

Comment:
the autoplay for vimeo doesn't seem to really work 🤔 So maybe if it ends up not being possible we can not output anything extra at the end of the url?
<details>
<summary>Screenshot</summary>

![](https://screenshot.click/30-03-0ozlp-5bh2x.png)
</details>

Reply:
Let me know if you see a discrepancy happening though. The one part of the shared string that vimeo has no use for is the &playlist, which youtube needs to allow looping apparently. Vimeo just ignores this.